### PR TITLE
Remove obsolete special paths for Missing

### PR DESCRIPTION
**Pull request by nalimilan**
These no longer make any difference to generated code.

### CODE REVIEW

File: src/dataframerow/dataframerow.jl [hunk 1]
```diff
@@ -62,11 +62,13 @@ Base.values(r::DataFrameRow) = ntuple(col -> parent(r)[col][row(r)], length(r))
 
 # hash column element
 Base.@propagate_inbounds hash_colel(v::AbstractArray, i, h::UInt = zero(UInt)) = hash(v[i], h)
-Base.@propagate_inbounds hash_colel(v::AbstractCategoricalArray, i, h::UInt = zero(UInt)) =
```
Comment (bkamins) [inline src/dataframerow/dataframerow.jl]: why is this not needed any more?

File: src/dataframerow/dataframerow.jl [hunk 2]
```diff
@@ -62,11 +62,13 @@ Base.values(r::DataFrameRow) = ntuple(col -> parent(r)[col][row(r)], length(r))
...
-    hash(CategoricalArrays.index(v.pool)[v.refs[i]], h)
 Base.@propagate_inbounds function hash_colel(v::AbstractCategoricalArray{>: Missing}, i, h::UInt = zero(UInt))
     ref = v.refs[i]
-    ref == 0 ? hash(missing, h) : hash(CategoricalArrays.index(v.pool)[ref], h)
+    if eltype(v) >: Missing && ref == 0
+        hash(missing, h)
+    else
+        hash(CategoricalArrays.index(v.pool)[ref], h)
```
Comment (bkamins) [inline src/dataframerow/dataframerow.jl]: I understand that here `ref` will never be `0` right?

Comment (nalimilan) [reply src/dataframerow/dataframerow.jl]: Yes (at least if the array isn't corrupt). This is really equivalent to the previous code, except that the type check is inside the function now.

File: src/dataframerow/dataframerow.jl (hunk 1 above)
Comment (nalimilan) [reply src/dataframerow/dataframerow.jl]: Woops, I forgot to widen the signature of the following method.

File: src/abstractdataframe/abstractdataframe.jl [hunk 3]
```diff
@@ -528,18 +528,17 @@ end
 ##############################################################################
 
 function _nonmissing!(res, col)
-    # workaround until JuliaLang/julia#21256 is fixed
-    eltype(col) >: Missing || return
-
     @inbounds for (i, el) in enumerate(col)
         res[i] &= !ismissing(el)
     end
+    return nothing
```
[the remark this answers is not in the document]
Comment (nalimilan) [reply src/abstractdataframe/abstractdataframe.jl]: Yes (I've checked that it's indeed the case).

**Pull request merged**